db/migrate/20240214190849_add_reservable_items_count_to_item_pools.rb

```diff
@@ -0,0 +1,5 @@
+class AddReservableItemsCountToItemPools < ActiveRecord::Migration[7.1]
+  def change
+    add_column :item_pools, :reservable_items_count, :integer, default: 0, null: false
```

Comment:
❓How is this field maintained? I see one reference to it in `ItemPool#max_reservable_quantity` but I don't see anything manipulating the field.

(Pretty sure this is working and it's just an issue with my understanding, so consider this non-blocking.)

Reply:
It's Rails Magic™! By specifying [`counter_cache: true` in the association declaration within `ReservableItem`](https://github.com/chicago-tool-library/circulate/pull/1327/files#diff-a99705a6be1ae926d7cdf2f6713144c1ae48013c65d1f029c97d50d9538ee2c2R3), Rails will [automatically update this count for us](https://guides.rubyonrails.org/association_basics.html#options-for-belongs-to-counter-cache).

As soon as we need to filter which associated items are counted (say, maybe we only want to count items with a certain status) we will have to implement it ourselves. But for now, this will do.